README.md

```diff
@@ -4,152 +4,293 @@
+```
+
+In all three cases you'll get a similar response parsed into a Clojure map:
+```clojure
+=>
+{:ok true
+ :result {:message_id 10
+          :from {:id 123, :is_bot true, :first_name "mybot", :username "my_roboto"}
+          :chat {:id 280000000, :first_name "Bill", :last_name "Howe", :username "myusername", :type "private"}
+          :date 1602816282
+          :text "Oh hi yourself."}}
+```
+
+Notice the incrementing `message_id` number. This is useful for keeping track of the message order and which messages
+have already been processed. Save it and use it later as the `get-updates` optional `offset` parameter.
+
+## Advanced Use
+
+### Error Handling
 
-;; Option 3: send required parameters as simple values, optional values as a map
-(tbot/send-message mybot 789 "oh hi yourself." {:disable_notification true})
+The library signals unsuccessful Bot API requests and other arbitrary errors during the Bot API calls.
 
-;; response - parsed into a Clojure map
-{:ok true, :result {:message_id 10, :from {:id 123, :is_bot true, :first_name "mybot", :username "my_roboto"}, :chat {:id 789, :first_name "Bill", :last_name "Howe", :username "myusername", :type "private"}, :date 1602816282, :text "oh hi yourself."}}
+```clojure
+(tbot/set-webhook mybot nil)
+=>
+{:ok false,
+ :error_code 400,
+ :description "Bad Request: bad webhook: Failed to resolve host: No address associated with hostname",
+ :error #error{...}}
 ```
 
+Upon any exception, it is passed either:
+- along with the Bot API response (in a form of Clojure map) in case the request was unsuccessful
+[in terms of the Bot API](https://core.telegram.org/bots/api/#making-requests), or
+- just as `{:error ex}` map in all other situations when we're unable to retrieve a response body.
+
+It's up to you to decide on how to react to an error in each particular case.
+
+The usual choices are:
+- to just log (w/ `:error_code` and `:description`), or
+- to retry (if the `:parameters` as present), or
+- to rethrow the `:error` object.
+
+> **NOTE:** In asynchronous code, where callbacks can be/are processed on a different thread
+> there is no sense in rethrowing exceptions. See the next section for details on how to make
+> an async Telegram Bot API request with the `telegrambot-lib`.
+
+### Making Asynchronous Calls
+
+Basically, you have to pass an `:async true` parameter along the bot instance to make any function end up
+with firing an asynchronous request. The only difference is that the function returns immediately, with a
+regular `core.async` channel to take the response (or [error result](#error-handling)) from later on.
+
+Here's how such a call may look like:
+
+```clojure
+;; Option 1: temporarily modify the bot instance for a particular async request
+(tbot/send-message (assoc mybot :async true)
+                   {:chat_id 280000000 :text "Drop-and-run!"})
+
+;; Option 2: use a global/shared bot instance for all asynchronous requests
+(def my-async-bot (-> (tbot/create)
```

Comment:
Not for 1.0.0, but a future release could consider making the :async true/false flag a first class parameter to the bot create fn. (default of false)

Reply:
Well, I am currently considering introducing pooling capabilities into the `telegrambot-lib.http` ns, and augmentation of the bot create fn makes a lot of sense in this context. We do not use one in the moment, but we should. Not only will it help improve performance and resource utilization efficiency, but it will also allow us/end user to fine tune the whole thing (sync/async, connect timeout, # of threads, etc.). I think I'll create a separate issue for that.